yew-functional-macro/tests/functional_macro/with-props-pass.rs

```diff
@@ -0,0 +1,21 @@
+#![no_implicit_prelude]
+
+use yew::prelude::*;
+use yew_functional::{functional_component};
+
+#[derive(Clone, Properties, PartialEq)]
+struct Props {
+    a: usize,
+}
+
+#[functional_component(Comp)]
+fn comp(props: &Props) -> Html {
+    html! {
+        <p>
+            { props.a }
+        </p>
+    }
+}
+
+
+fn main() {}
```

Comment:
The `no_implicit_prelude` tests are most effective when we don't bring anything into scope.
For instance, right now this test wouldn't have caught the unqualified `Html` type the initial version had.

Normally you can just define the test setup in a submodule, but for the `functional_component` attribute that can only really be done for the `Props`.

```suggestion
#![no_implicit_prelude]

#[derive(Clone, PartialEq, ::yew::Properties)]
struct Props {
    a: usize,
}

#[::yew_functional::functional_component(Comp)]
fn comp(props: &Props) -> ::yew::Html {
    ::yew::html! { <p>{ props.a }</p> }
}

fn main() {}
```

Reply:
We could also use a test for the `ref props: &Props` pattern, just to make sure that actually works now.

Reply:
I think we should just remove `no_implicit_prelude`. Changing it to how you suggested makes the code too verbose. Or is there any reason to keep this?

Reply:
The reason for this is to test the hygiene of the macro. It's very easy for procedural macros to accidentally use items from the local scope (the original version of this PR had one such error, for example). Needless to say that that isn't a good idea; dealing with the resulting errors can be very confusing for the user and annoying to fix for the developers so I would like to ensure we're guarded against this going forward.

Yes, it makes the passing tests a bit more verbose, but this is an integral part of the test in the first place.
It only needs to be done for the passing tests so it really shouldn't be that bad.

Reply:
I've changed the two passing tests to use it.